From 13d8c9ee4dc1142f34898cac029daecbabfed2ce Mon Sep 17 00:00:00 2001
From: Xie Haocheng <haocheng.xie@amd.com>
Date: Wed, 30 Nov 2022 14:33:37 +0800
Subject: [PATCH 0405/1646] [Backport] kabi: Fix kabi breakage caused by new
 member added to struct perf_sample_data.

amd inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I601MP

Reference: https://gitee.com/openeuler/kernel/commit/8c08ad8400c4960cb4cdc1d451f7a9200c508bdd

--------------------------------

The kabi breakage was caused by the commit below:
3aac580d5cc3 perf: Add sample_flags to indicate the PMU-filled sample data
Use KABI_BROKEN_INSERT(u64 sample_flags) to remove the kabi error explicitly.

Signed-off-by: Xie Haocheng <haocheng.xie@amd.com>
Signed-off-by: Lin Yujun <linyujun809@huawei.com>
---
 include/linux/perf_event.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/perf_event.h b/include/linux/perf_event.h
index 3dcfc9a902b9..f0a55bc3f3d7 100644
--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -1006,7 +1006,6 @@ struct perf_sample_data {
 	 * Fields set by perf_sample_data_init(), group so as to
 	 * minimize the cachelines touched.
 	 */
-	u64				sample_flags;
 	u64				addr;
 	struct perf_raw_record		*raw;
 	struct perf_branch_stack	*br_stack;
@@ -1041,6 +1040,7 @@ struct perf_sample_data {
 
 	u64				phys_addr;
 	u64				cgroup;
+	KABI_BROKEN_INSERT(u64 sample_flags)
 } ____cacheline_aligned;
 
 /* default value for data source */
-- 
2.17.1

